src/EditorFeatures/Text/Extensions.cs

```diff
@@ -37,7 +40,7 @@ public static SourceText AsText(this ITextSnapshot textSnapshot)
             => SnapshotSourceText.From(textSnapshot);
 
         internal static SourceText AsRoslynText(this ITextSnapshot textSnapshot, Encoding encoding)
-            => new SnapshotSourceText.ClosedSnapshotSourceText(textSnapshot, encoding);
+            => new SnapshotSourceText.ClosedSnapshotSourceText(((ITextSnapshot2)textSnapshot).TextImage, encoding);
```

Comment:
I recognize this isn't your doing, but why are we creating ClosedSnapshotSourceTexts here? How do we know it's closed...? Is this just broken?